stacker/dag/__init__.py

```diff
@@ -0,0 +1,313 @@
+import threading
+import logging
+from copy import copy, deepcopy
+from collections import deque
+
+logger = logging.getLogger(__name__)
+
+try:
+    from collections import OrderedDict
+except:
+    from ordereddict import OrderedDict
+
+
+class DAGValidationError(Exception):
+    pass
+
+
+class DAG(object):
```

Comment:
FYI, this implementation is pulled from https://github.com/thieman/py-dag with a few changes (adds `walk` and `transpose` methods). Since it's small, I just pulled it in directly so we don't need to add an extra dependency.